modules/carto/src/layers/h3-tileset-2d.ts

```diff
@@ -66,12 +66,21 @@ function tileToBoundingBox(index: string): GeoBoundingBox {
 // similar
 // Relative scale factor (0 = no biasing, 2 = a few hexagons cover view)
 const BIAS = 2;
-export function getHexagonResolution(viewport): number {
-  const hexagonScaleFactor = (2 / 3) * viewport.zoom;
+export function getHexagonResolution(
+  viewport: {zoom: number; latitude: number},
+  tileSize: number
+): number {
+  // Difference in given tile size compared to deck's internal 512px tile size,
+  // expressed as an offset to the viewport zoom.
+  const zoomOffset = Math.log2(tileSize / 512);
+  const hexagonScaleFactor = (2 / 3) * (viewport.zoom - zoomOffset);
   const latitudeScaleFactor = Math.log(1 / Math.cos((Math.PI * viewport.latitude) / 180));
 
   // Clip and bias
-  return Math.max(0, Math.floor(hexagonScaleFactor + latitudeScaleFactor - BIAS));
+  return Math.max(
+    0,
+    Math.floor(hexagonScaleFactor + latitudeScaleFactor - (BIAS + zoomOffset / 3))
```

Comment:
This doesn't seem right to me, `zoomOffset` shouldn't be included in the final formula twice. If we simplify the equation we get:

```
Math.floor((2 / 3) * (viewport.zoom - zoomOffset) + latitudeScaleFactor - (BIAS + zoomOffset / 3));
simpifies to =>
Math.floor((2 / 3) * (viewport.zoom) + latitudeScaleFactor - (BIAS + zoomOffset);
```

I was imagining that the 1 / 3 offset would be applied to the constant, i.e. `const BIAS = 2 + 1 / 3` when we switch from `tileSize` 512 to 1024.

Reply:
The way I understand it is that when we switch the default `tileSize`, the maps shouldn't change. Separately, when the user updates `tileSize` in their code, the on-screen size should be correct.

If the h3 resolution is proportional to `zoomOffset` the on-screen size will not be correct

Reply:
(demo shared privately)

> ... when we switch the default tileSize, the maps shouldn't change.

This is my goal as well, but I don't think it's compatible with constant bias. The current logic keeps maps from changing at tile sizes 256, 512, and 1024. Adjusting bias with a mod operator instead of a multiplier would work just as well, for those specific tile sizes...

```javascript
// (a)
const bias = BIAS + zoomOffset / 3;

// (b)
const bias = BIAS + (zoomOffset % 2) / 3;
```

...but I haven't worked out an equivalent fix for tileSize=2048 or other values yet, which feels important. My feeling is that if "maps shouldn't change" is a requirement, we may need to expose the bias as a prop.